src/components/tree/phyloTree/layouts.js

```diff
@@ -288,6 +289,57 @@ export const setDistance = function setDistance(distanceAttribute) {
   timerEnd("setDistance");
 };
 
+/**
+ * given nodes add y values (node.displayOrder) to every node
+ * Nodes are the phyloTree nodes (i.e. node.n is the redux node)
+ * Nodes must have parent child links established (via createChildrenAndParents)
+ * PhyloTree can subsequently use this information. Accessed by prototypes
+ * rectangularLayout, radialLayout, createChildrenAndParents
+ * side effects: node.displayOrder and node.displayOrderRange (i.e. in the phyloTree node)
+ */
+export const calcYValues = (nodes, focus) => {
```

Comment:
Brief comments (after looking at the code for perhaps the first time since I wrote the original prototype!):

1. We should be able to get rid of `setDisplayOrder` as it's essentially doing the same thing as `calcYValues` and just use one function throughout the code.
2. `setDisplayOrder` specifically considers the case of an exploded tree which made me wonder what happens if we explode a tree and then toggle on focus - it's a bug! We should test the similar case of a JSON with n>1 trees (subtrees). In either case, for v1 functionality we can disable the focus functionality when we've exploded a tree / have multiple subtrees in the JSON.
3. Now that this is called frequently (rather than simply on dataset load) we can probably make it faster if needed. I see `setTreeFocus` has timers in it - we should build Auspice with `--includeTiming` and see how fast/slow this currently is.

Reply:
1. I chose to drop `calcYValues` and add focus functionality to `setDisplayOrder`: 9a2bdb451f3c09046ab0341bb946d7080953ab33
2. Added to checklist:

    > Disable toggle and turn off focus when using exploded tree / multiple trees in JSON

3. I did this locally and through some spot checking only saw `Timer setTreeFocus (#XX) took 0ms. Average: 0ms.`. Maybe I'm not checking in the right places though. Maybe we can tweak the review app to build with `--includeTiming`?

Reply:
That looks great! Yes ideally we'd allow it, I just wanted to convey it wasn't essential for the v1.

Reply:
Re: timing (3)

I've updated https://github.com/nextstrain/nextstrain.org/pull/1025 with https://github.com/nextstrain/nextstrain.org/pull/1025/commits/0671e90b0cdfa55d0ee5b01d0b78bc933cedb121 so the review app has timing messages.

The function is now `setDisplayOrder`. Playing with various date range filters on [this n=3343](https://nextstrain-s-nextstrain-l8gysc.herokuapp.com/ncov/gisaid/global/all-time@2024-10-14) dataset, the function takes between 0-3ms for me (M1 mac, Chromium browser):

<img width="672" alt="image" src="https://github.com/user-attachments/assets/0d23b767-25f0-4228-8118-2f11cbbbe790">

Reply:
Those times are fine - they make it clear it's not something we need to optimise at the moment. Thanks for looking into it!

Reply:
Thanks for the pointer! I'll still plan to keep 330e87d43ab459822ae1584036e4c788ee5369d9 in the PR.